[libzip] update to rel-1-6-1

mitzal opened the pull request:
Libzip updated to rel-1-6-1
[ #10778 ]

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=10784) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=10784) before we can accept your contribution.<br/><br/>:x: mitzal [sign now](https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=10784)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/microsoft/vcpkg?pullRequest=10784) it.</sub>

LilyWangL commented on ports/libzip/portfile.cmake:
```diff
@@ -1,10 +1,9 @@
-	PATCHES avoid_computation_on_void_pointer.patch
 )
 
 # AES encryption
```
```
set(USE_OPENSSL OFF)
if("openssl" IN_LIST FEATURES)
    set(USE_OPENSSL ON)
endif()
set(USE_BZIP2 OFF)
if("bzip2" IN_LIST FEATURES)
    set(USE_BZIP2 ON)
endif()
```
The way to handle feature can be replaced by [vcpkg_check_features()](https://github.com/microsoft/vcpkg/blob/a3a6530631df905eb5c0e26d0b20d7d548e0c465/docs/maintainers/vcpkg_check_features.md).

mitzal replied:
changed

mitzal committed:
Libzip updated to rel-1-6-1

janisozaur commented on ports/libzip/portfile.cmake:
```diff
@@ -1,34 +1,26 @@
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
     OPTIONS
+        ${FEATURE_OPTIONS}
```
The options are [`ENABLE_OPENSSL` and `ENABLE_BZIP2` in libzip's code](https://github.com/nih-at/libzip/blob/5487dcae97291c5eb598ea129aff045e1e2eeb65/CMakeLists.txt#L13-L16), but I see no mapping here between `OPENSSL` and `ENABLE_OPENSSL`. @mitzal is that intended and correct?

mitzal replied:
Should be ENABLE_OPENSSL, you're right; thanks for changes.

janisozaur committed:
[libzip] Add windows_crypto feature

The new BCrypt/CNG-based crypto uses BCryptDeriveKeyPBKDF2 which is only
available since WinNT 6.1. It is important to me as a consumer of libzip
through vcpkg to be able to turn this off.

janisozaur reviewed (changes requested):
Please merge https://github.com/mitzal/vcpkg/pull/1

mitzal committed:
Merge pull request #1 from janisozaur/libzip-windows-crypto

[libzip] Add windows_crypto feature

ras0219-msft commented:
Looking at the upstream code[1], it seems like only one of the crypto backends will be used. Therefore, it seems to me that we should restructure the features as:
```
Default-Features: default-aes
...
Feature: default-aes
Build-Depends: libzip[core,openssl] (!windows), libzip[core,wincrypto] (windows)

Feature: openssl
Build-Depends: openssl

Feature: wincrypto
```

We also need to explicitly disable the other backends (mbedTLS, gnutls, and commoncrypto) to ensure they aren't accidentally linked. These could be optionally made into features in the future if desired.

[1] https://github.com/nih-at/libzip/blob/8bd4f66d906026ab85bd2d05e47c6be0e144135f/CMakeLists.txt#L259

mitzal commented:
> Looking at the upstream code[1], it seems like only one of the crypto backends will be used. Therefore, it seems to me that we should restructure the features as:
> 
> ```
> Default-Features: default-aes
> ...
> Feature: default-aes
> Build-Depends: libzip[core,openssl] (!windows), libzip[core,wincrypto] (windows)
> 
> Feature: openssl
> Build-Depends: openssl
> 
> Feature: wincrypto
> ```
> 
> We also need to explicitly disable the other backends (mbedTLS, gnutls, and commoncrypto) to ensure they aren't accidentally linked. These could be optionally made into features in the future if desired.
> 
> [1] https://github.com/nih-at/libzip/blob/8bd4f66d906026ab85bd2d05e47c6be0e144135f/CMakeLists.txt#L259

Features have been restructured and some additional options have been added.

mitzal committed:
Additional options added.

janisozaur commented:
https://dev.azure.com/vcpkg/public/_build/results?buildId=34181&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a&t=a4822bc1-f3a9-5984-c941-6c1b7aa2ab81&l=18

`|libzip   |Fail|core|Test passes in baseline but fails in current run. If expected update ci.baseline.txt with 'libzip:x64-linux=fail'`

mitzal committed:
Common Crypto supported as an option for encryption backend.

ras0219-msft commented:
LGTM, thanks for the fixes!
